components/abstractions/src/main/java/com/microsoft/kiota/Headers.java

```diff
@@ -31,19 +31,36 @@ protected Headers(@Nonnull Headers headers) {
      * @param value the value of the header to add.
      */
     public void add(@Nonnull final String key, @Nonnull final String value) {
+        addImpl(key, value, true);
+    }
+
+    /**
+     * Adds a header to the current request if it was not already set
+     *
+     * @param key   the key of the header to add.
+     * @param value the value of the header to add.
+     */
+    public void tryAdd(@Nonnull final String key, @Nonnull final String value) {
```

Comment:
maybe this should return a boolean to indicate whether or not the add was successful? (at least it's done this way in dotnet)